eo-maven-plugin/src/test/java/org/eolang/maven/testapi/FakeMaven.java

```diff
@@ -0,0 +1,195 @@
+
+/**
+ * Fake maven workspace that executes Mojos in order to test
+ * their behaviour and results.
+ * @since 0.28.12
+ * @todo #1417:90min Make `FakeMaven.exec()` return a `HashMap` with all
+ *  files created in the directory and their relative names. Then, we can assert on this hash map.
+ */
+public final class FakeMaven {
+
+    /**
+     * Default eo program id.
+     */
+    private static final String PROGRAM_ID = "foo.x.main";
+
+    /**
+     * Default eo program path.
+     */
+    private static final String PROGRAM_PATH = "foo/x/main.eo";
+
+    /**
+     * Default eo-foreign.csv file format.
+     */
+    private static final String FOREIGN_FORMAT = "csv";
+
+    /**
+     * Default eo-foreign.csv file path.
+     */
+    private static final Path FOREIGN_PATH = Paths.get("eo-foreign.csv");
+
+    /**
+     * Test workspace where we place all programs, files, compilation results, etc.
+     */
+    private final Home workspace;
+
+    /**
+     * Mojos params.
+     */
+    private final Map<String, Object> params;
+
+    /**
+     * Attributes for eo.foreign.*.
+     */
+    private final Map<String, Object> attributes;
+
+    /**
+     * Path to a program in workspace.
+     */
+    private Path prog;
+
+    /**
+     * The main constructor.
+     *
+     * @param workspace Test temporary directory.
+     */
+    public FakeMaven(final Path workspace) {
+        this.workspace = new Home(workspace);
+        this.params = this.defaultParams();
+        this.attributes = new HashMap<>();
+    }
+
+    /**
+     * Adds eo program to a workspace.
+     * @param program Program as a raw string.
+     * @return Workspace with eo program.
+     * @throws IOException If can't save eo program in workspace.
+     */
+    public FakeMaven forProgram(final String... program) throws IOException {
+        final Path path = Paths.get(FakeMaven.PROGRAM_PATH);
+        this.workspace.save(String.join("\n", program), path);
+        this.prog = path;
+        return this;
+    }
+
+    /**
+     * Sets parameter for execution.
+     *
+     * @param key Parameter name
+     * @param value Parameter value
+     * @return The same maven instance.
+     */
+    public FakeMaven with(final String key, final Object value) {
+        this.params.put(key, value);
+        return this;
+    }
+
+    /**
+     * Sets tojo attribute.
+     *
+     * @param attribute Tojo attribute.
+     * @param value Attribute value.
+     * @return The same maven instance.
+     */
+    public FakeMaven with(final TojoAttribute attribute, final Object value) {
+        this.attributes.put(attribute.toString(), value);
+        return this;
+    }
+
+    /**
+     * Executes Mojo in the workspace.
+     *
+     * @param mojo Mojo to execute.
+     * @param <T> Template for descendants of Mojo.
+     * @return Workspace after executing Mojo.
+     */
+    public <T extends AbstractMojo> FakeMaven execute(final Class<T> mojo) {
+        this.withEoForeign();
```

Comment:
@volodya-lombrozo this also looks wrong to me. Let's make our users call this. Now it's happening behind the scene and it's misleading.

Reply:
@yegor256 Done. 
___
But I disagree, because you expose implementation (again, it's about encapsulation):
1. afaik, all tests will require `eo-foreign.csv`. What is the point to call it each time in each test the same way?
2. You are getting attached to the specific implementation through `eo-foreign` file, what if we decide to use a different approach? It will difficult to fix all that tests.